### PR TITLE
memcheck: Add QF path and name to error message

### PR DESCRIPTION
**Pull request by jrwrigh**
This makes is much easier to determine where the memory violation is detected. Example of what the error message looks like in fluids:
```
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: Error in external library
[0]PETSC ERROR: /home/jrwrigh/software/libCEED/backends/memcheck/ceed-memcheck-qfunction.c:70 in CeedQFunctionApply_Memcheck(): QFunction output 2 entry 0 is NaN after restoring write-only access: /home/jrwrigh/software/libCEED/examples/fluids/problems/../qfunctions/advection2d.h:IFunction_Advection2d
[0]PETSC ERROR: See https://petsc.org/release/faq/ for trouble shooting.
[0]PETSC ERROR: Petsc Development GIT revision: v3.20.2-298-ga5e892f4d25  GIT Date: 2023-12-12 15:52:45 -0700
```

Open to suggestions on what the error message should look like.

### CODE REVIEW

File: backends/memcheck/ceed-memcheck-qfunction.c [hunk 1]
```diff
@@ -63,8 +63,12 @@ static int CeedQFunctionApply_Memcheck(CeedQFunction qf, CeedInt Q, CeedVector *
     // Note: need field size because vector may be longer than needed for output
     CeedCallBackend(CeedQFunctionFieldGetSize(output_fields[i], &field_size));
     for (CeedSize j = 0; j < Q * field_size; j++) {
-      CeedCheck(!isnan(impl->outputs[i][j]), ceed, CEED_ERROR_BACKEND, "QFunction output %d entry %ld is NaN after restoring write-only access", i,
-                j);
+      char *kernel_path, *kernel_name;
+      CeedCallBackend(CeedQFunctionGetSourcePath(qf, &kernel_path));
+      CeedCallBackend(CeedQFunctionGetKernelName(qf, &kernel_name));
```
Comment (jeremylt) [inline backends/memcheck/ceed-memcheck-qfunction.c:68]: This should be outside of the loop

Comment (jrwrigh) [inline backends/memcheck/ceed-memcheck-qfunction.c:68]: Also not sure how much I like where this code is placed. But those char pointers are really only needed here.

Comment (jrwrigh) [reply backends/memcheck/ceed-memcheck-qfunction.c:68]: Jinx. I couldn't decide where to place them so I just threw them there.

File: backends/memcheck/ceed-memcheck-qfunction.c [hunk 2]
```diff
@@ -63,8 +63,12 @@ static int CeedQFunctionApply_Memcheck(CeedQFunction qf, CeedInt Q, CeedVector *
     // Note: need field size because vector may be longer than needed for output
     CeedCallBackend(CeedQFunctionFieldGetSize(output_fields[i], &field_size));
     for (CeedSize j = 0; j < Q * field_size; j++) {
-      CeedCheck(!isnan(impl->outputs[i][j]), ceed, CEED_ERROR_BACKEND, "QFunction output %d entry %ld is NaN after restoring write-only access", i,
-                j);
+      char *kernel_path, *kernel_name;
```
Comment (jeremylt) [inline backends/memcheck/ceed-memcheck-qfunction.c:66]: hmm, now I'm wondering if these should be `const` parameters

File: backends/memcheck/ceed-memcheck-qfunction.c (hunk 1 above)
Comment (jeremylt) [reply backends/memcheck/ceed-memcheck-qfunction.c:68]: Ha, You could do 

```
  {
    char *foo, *bar;

    // Get the ptrs
    for (....) {
    }
  }
```

File: backends/memcheck/ceed-memcheck-qfunction.c [hunk 3]
```diff
@@ -56,18 +56,23 @@ static int CeedQFunctionApply_Memcheck(CeedQFunction qf, CeedInt Q, CeedVector *
...
-    CeedInt field_size;
-
-    // Note: need field size because vector may be longer than needed for output
-    CeedCallBackend(CeedQFunctionFieldGetSize(output_fields[i], &field_size));
-    for (CeedSize j = 0; j < Q * field_size; j++) {
-      CeedCheck(!isnan(impl->outputs[i][j]), ceed, CEED_ERROR_BACKEND, "QFunction output %d entry %ld is NaN after restoring write-only access", i,
-                j);
+  {
+    char *kernel_path, *kernel_name;
+    CeedCallBackend(CeedQFunctionGetSourcePath(qf, &kernel_path));
```
Comment (jeremylt) [inline backends/memcheck/ceed-memcheck-qfunction.c:61]:
```suggestion
    char *kernel_path, *kernel_name;
    
    CeedCallBackend(CeedQFunctionGetSourcePath(qf, &kernel_path));
```
minor - one line between declarations and code

File: backends/memcheck/ceed-memcheck-qfunction.c (hunk 1 above)
Comment (jrwrigh) [reply backends/memcheck/ceed-memcheck-qfunction.c]: Pushed up. I put it outside of both for loops

**Review by jeremylt — APPROVED** [review]

File: backends/memcheck/ceed-memcheck-qfunction.c (hunk 3 above)
Comment (jrwrigh) [reply backends/memcheck/ceed-memcheck-qfunction.c]: fixed.

File: backends/memcheck/ceed-memcheck-qfunction.c (hunk 2 above)
Comment (jrwrigh) [reply backends/memcheck/ceed-memcheck-qfunction.c]: Yeah, I'd probably say so. I can't think of any reason that they should ever need to change after the QFunction has been created.

Comment (jeremylt) [reply backends/memcheck/ceed-memcheck-qfunction.c]: cool, I'll investigate in a follow-up

**Pull request merged**